yarpl/include/yarpl/flowable/EmitterFlowable.h

```diff
@@ -164,8 +161,12 @@ class EmiterSubscription : private Subscription, private Subscriber<T> {
       int64_t emitted;
       bool done;
 
-      std::tie(emitted, done) =
-          emiter_->emit(get_ref<Subscriber<T>>(this), current);
+      std::tie(emitted, done) = emiter_->emit(this_subscriber, current);
+
+      if(this->count() == 1) {
```

Comment:
The `while` loop is just there to satisfy a race around the compare&swap with the `requested_` member, if multiple threads have a handle to us and are requesting (eg count() > 1). So this is safe to do here rather than the while loop, as if count() == 1, we'll only iterate through the loop once (also, we should be done/canceled if nobody has a reference to us, else it's a bug).